src/server/services/afis/afis-facturen.ts

```diff
@@ -260,15 +260,43 @@ function transformFacturen(
 ): AfisFacturenResponse {
   const feedProperties = getFeedEntryProperties(responseData);
   const count = responseData?.feed?.count ?? feedProperties.length;
-  const facturenTransformed = feedProperties.map((invoiceProperties) => {
-    return transformFactuur(invoiceProperties, sessionID, deelbetalingen);
-  });
+  const facturenTransformed = feedProperties
+    .filter((invoiceProperties) => {
+      return shouldFilterOutFactuur(invoiceProperties.PostingDate);
+    })
+    .map((invoiceProperties) => {
+      return transformFactuur(invoiceProperties, sessionID, deelbetalingen);
+    });
   return {
     count,
     facturen: facturenTransformed,
   };
 }
 
+function shouldFilterOutFactuur(postingDate: string): boolean {
+  const postingDateAsDate = new Date(postingDate);
+  const now = new Date();
+
+  if (postingDateAsDate.getTime() > now.getTime()) {
+    captureMessage('postingDate of invoice is in the future', {
+      severity: 'error',
+    });
+    return false;
+  }
+
+  const postedToday =
+    postingDateAsDate.getDate() === now.getDate() &&
+    postingDateAsDate.getMonth() === now.getMonth() &&
+    postingDateAsDate.getFullYear() === now.getFullYear();
+
+  if (!postedToday) {
+    return true;
+  }
+
+  const sevenOClock = 19;
+  return postedToday && now.getHours() >= sevenOClock;
```

Comment:
```suggestion
  return isToday(datePosted) && isAfter(datePosted, dateTodayAvailableAfterOClock)
```

Maar dan zou je ook iets kunnen doen als:

```
  const isPastDate = !isToday(datePosted) && isPast(datePosted);
  const isTodayAfterDownloadOclock =
    isToday(datePosted) && isAfter(datePosted, dateTodayAvailableAfterOClock);

  return isPastDate || isTodayAfterDownloadOclock;
```

Heb je geen future check nodig.

Reply:
Alright even kijken

Reply:
oeeee ja heel nice

Reply:
Ah whoops test falen nog, hmmm